src/eval.rs

```diff
@@ -1712,7 +1712,17 @@ fn apply_continuation<F: LurkField>(
                                     store.get_u64((a / b).into())
                                 }
                             }
-                            Op2::Modulo => store.get_u64((a % b).into()),
+                            Op2::Modulo => {
+                                if b.is_zero() {
+                                    return Ok(Control::Return(
+                                        *result,
+                                        *env,
+                                        store.intern_cont_error(),
```

Comment:
This will need to be updated to use the error mechanism introduced in #201.

Reply:
That's correct, but refactoring `apply_continuation`'s control flow to return an enum including error cases and then to convert this to a `Result` in a second phase (as done with `reduce_with_witness` in #224) :
- would extend over 600 SLoC, vs. this 12-line fix,
- seems topically different from fixing the `mod` / `div` discrepancy issue of #197. 

Shall we open an issue (e.g. from this comment) and tackle this in a separate PR?

Reply:
Ah, yeah… I missed that this was in `apply_continuation` so would require all that. I agree with your proposal.